model/cicd/entities.yaml

```diff
@@ -24,6 +24,12 @@ groups:
     name: cicd.worker
     brief: >
       A CICD worker is a component of the CICD system that performs work (eg. running pipeline tasks or performing sync).
+
+      A single pipeline run may be distributed across multiple workers. Each span representing a pipeline run or a task run
+      should be linked to the worker that performed the corresponding work.
+
+      When a pipeline run involves several workers, its task run spans may reference different `cicd.worker` resources.
+      The pipeline run's parent span may instead reference the CICD controller as the `cicd.worker` resource.
```

Comment:
Should span relationship be documented on the spans? any reason to document them on entities?

Reply:
Usually the tasks executing on a worker emit several spans. By setting the worker as an entity it's more efficient than setting the worker attributes on each span.

Reply:
Thanks for the clarification.

> by linked you mean associated with the entity, please use this wording - links between spans are something different

Agreed, I have changed this as requested: 4cf970096841e55d9deb2c8243a0eb7b52693c8c

> should be part of span docs

While I wrote here about spans I think the clarification for the worker description regarding a pipeline run executing across multiple workers is applicable to more than just spans.
It would also apply to any logs of a pipeline run emitted on each worker or the controller (as well as metrics, cf #1184).
As such I think we should keep this in the worker entities description.

I will have to think about making this description more generic...

Reply:
> I will have to think about making this description more generic

Done: 5d721b52f696793aebae441924a7dd90c71a7d7e and 9cc63f949484015e8e979284c598c9df142b8882